app/scripts/lib/segment/analytics.js

```diff
@@ -0,0 +1,212 @@
+import removeSlash from 'remove-trailing-slash';
+import looselyValidate from '@segment/loosely-validate-event';
+import { isString } from 'lodash';
+
+const noop = () => ({});
+
+// Taken from https://stackoverflow.com/a/1349426/3696652
+const characters =
+  'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789';
+const generateRandomId = () => {
+  let result = '';
+  const charactersLength = characters.length;
+  for (let i = 0; i < 20; i++) {
+    result += characters.charAt(Math.floor(Math.random() * charactersLength));
+  }
+  return result;
+};
+
+export default class Analytics {
+  /**
+   * Initialize a new `Analytics` with Segment project's `writeKey` and an
+   * optional dictionary of `options`.
+   *
+   * @param {string} writeKey
+   * @param {object} [options] - (optional)
+   * @property {number} [flushAt] (default: 20)
+   * @property {number} [flushInterval] (default: 10000)
+   * @property {string} [host] (default: 'https://api.segment.io')
+   */
+  constructor(writeKey, options = {}) {
+    this.writeKey = writeKey;
+
+    this.host = removeSlash(options.host || 'https://api.segment.io');
+    this.flushInterval = options.flushInterval || 10000;
+    this.flushAt = options.flushAt || Math.max(options.flushAt, 1) || 20;
+
+    this.queue = [];
+    this.path = '/v1/batch';
+    this.maxQueueSize = 1024 * 450;
+    this.flushed = false;
+
+    Object.defineProperty(this, 'enable', {
+      configurable: false,
+      writable: false,
+      enumerable: true,
+      value: true,
+    });
+  }
+
+  _validate(message, type) {
+    looselyValidate(message, type);
+  }
+
+  _message(type, message, callback) {
+    this._validate(message, type);
+    this.enqueue(type, message, callback);
+    return this;
+  }
+
+  /**
+   * Send an identify `message`.
+   *
+   * @param {object} message
+   * @param {Function} [callback] - (optional)
+   * @returns {Analytics}
+   */
+  identify(message, callback) {
+    return this._message('identify', message, callback);
+  }
+
+  /**
+   * Send a track `message`.
+   *
+   * @param {object} message
+   * @param {Function} [callback] - (optional)
+   * @returns {Analytics}
+   */
+  track(message, callback) {
+    return this._message('track', message, callback);
+  }
+
+  /**
+   * Send a page `message`.
+   *
+   * @param {object} message
+   * @param {Function} [callback] - (optional)
+   * @returns {Analytics}
+   */
+  page(message, callback) {
+    return this._message('page', message, callback);
+  }
+
+  /**
+   * Add a `message` of type `type` to the queue and
+   * check whether it should be flushed.
+   *
+   * @param {string} type
+   * @param {object} msg
+   * @param {Function} [cb] - (optional)
+   */
+  enqueue(type, msg, cb) {
+    const callback = cb || noop;
```

Comment:
We can remove this line if we default default callback to noop in the args

Reply:
done 👍